bottlerocket/agents/src/sonobuoy.rs

```diff
@@ -81,19 +85,75 @@ pub async fn rerun_failed_sonobuoy(
         .args(kubeconfig_arg.to_owned())
         .arg("run")
         .args(e2e_repo_arg)
-        .arg("--wait")
         .arg("--rerun-failed")
         .arg(results_filepath.as_os_str())
         .status()
         .context(error::SonobuoyProcessSnafu)?;
-    info!("Sonobuoy testing has completed, checking results");
 
     // TODO - log something or check what happened?
     ensure!(status.success(), error::SonobuoyRunSnafu);
 
+    info!("Sonobuoy testing has started, waiting for results");
+    wait_for_sonobuoy_results(kubeconfig_path, None).await?;
```

Comment:
I think we should add an enforeced timeout here with `tokio::time::timeout` otherwise we're relying on the sonobuoy test to eventually change out of the `running` status at some point which is not a guarantee.

Example here: https://github.com/bottlerocket-os/bottlerocket/blob/89c866f8d4e9ecaef55d6027d830baa2219e6051/sources/api/pluto/src/eks.rs#L115-L119

Reply:
`tokio::time::timeout` completely slipped my mind, so I had a failure counter in the loop :upside_down_face:
I'll add the `timeout` and remove the failure check.